### PR TITLE
elfutils: add patch that fixes compilation under gcc8

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

Was failing: https://hydra.nixos.org/build/73828348/nixlog/2

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/elfutils/default.nix [hunk 1]
```diff
@@ -10,7 +10,11 @@ stdenv.mkDerivation rec {
     sha256 = "0rp0r54z44is49c594qy7hr211nhb00aa5y7z74vsybbaxslg10z";
   };
 
-  patches = ./debug-info-from-env.patch;
+  patches = [
+    ./debug-info-from-env.patch
+    # From: https://patchwork.openembedded.org/patch/149575/
+    ./0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch
```
Comment (jtojnar) [inline pkgs/development/tools/misc/elfutils/default.nix]: Could you use
```nix
# From: https://patchwork.openembedded.org/patch/149575/
(fetchpatch {
  name = "0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch";
  url = https://patchwork.openembedded.org/patch/149575/raw/;
  sha256 = "0000000000000000000000000000000000000000000000000000000000000000";
})
```

Comment (Synthetica9) [reply pkgs/development/tools/misc/elfutils/default.nix]: Is that also possible if it's a patch within a patch? (The patch that is linked adds a patch file, it isn't just the patch)

Comment (Synthetica9) [reply pkgs/development/tools/misc/elfutils/default.nix]: Relevant part of the linked patch:
```
diff --git a/meta/recipes-devtools/elfutils/files/0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch b/meta/recipes-devtools/elfutils/files/0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch
new file mode 100644
index 0000000000..cf4d1dfe75
--- /dev/null
+++ b/meta/recipes-devtools/elfutils/files/0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch
@@ -0,0 +1,34 @@ 
+From 732913a8c35c7b25c0cbf6903cab1ad6b602b525 Mon Sep 17 00:00:00 2001
+From: Khem Raj <raj.khem@gmail.com>
+Date: Thu, 22 Mar 2018 22:44:03 -0700
+Subject: [PATCH] Ensure that packed structs follow the gcc memory layout
+
+Partial backport of
+https://sourceware.org/git/?p=elfutils.git;a=commit;h=17d7194d291bf91d130b78e06cbe27b290e0376d
+
+Helps fixing alignment errors e.g.
+linux-core-note.c:116:1: error: alignment 2 of 'struct m68k_prstatus
+' is less than 4 [-Werror=packed-not-aligned]
+ ;
+ ^
+
+Upstream-Status: Backport [https://sourceware.org/git/?p=elfutils.git;a=commit;h=17d7194d291bf91d130b78e06cbe27b290e0376d]
+
+Signed-off-by: Khem Raj <raj.khem@gmail.com>
+---
+ backends/linux-core-note.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/backends/linux-core-note.c b/backends/linux-core-note.c
+index 5f06c896..9faae4c3 100644
+--- a/backends/linux-core-note.c
++++ b/backends/linux-core-note.c
+@@ -111,7 +111,7 @@ struct EBLHOOK(prstatus)
+   FIELD (INT, pr_fpvalid);
+ }
+ #ifdef ALIGN_PRSTATUS
+-  __attribute__ ((packed, aligned (ALIGN_PRSTATUS)))
++  attribute_packed __attribute__ ((aligned (ALIGN_PRSTATUS)))
+ #endif
+ ;
+ 
```

Comment (jtojnar) [reply pkgs/development/tools/misc/elfutils/default.nix]: This URL then http://git.openembedded.org/openembedded-core/plain/meta/recipes-devtools/elfutils/files/0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch?id=49aae1d75ff1c6a9643c30a8cc5776a2ffa83dd3

**Comment by Synthetica9** [comment]
@jtojnar Better?

File: pkgs/development/tools/misc/elfutils/default.nix [hunk 2]
```diff
@@ -10,7 +10,14 @@ stdenv.mkDerivation rec {
     sha256 = "0rp0r54z44is49c594qy7hr211nhb00aa5y7z74vsybbaxslg10z";
   };
 
-  patches = ./debug-info-from-env.patch;
+  patches = [
+    ./debug-info-from-env.patch
+    (fetchpatch {
```
Comment (jtojnar) [inline pkgs/development/tools/misc/elfutils/default.nix]: `fetchurl` should be enough here, it is just a static file.

File: pkgs/development/tools/misc/elfutils/default.nix [hunk 3]
```diff
@@ -10,7 +10,14 @@ stdenv.mkDerivation rec {
...
+      name = "0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch";
+      url = http://git.openembedded.org/openembedded-core/plain/meta/recipes-devtools/elfutils/files/0001-Ensure-that-packed-structs-follow-the-gcc-memory-lay.patch?id=49aae1d75ff1c6a9643c30a8cc5776a2ffa83dd3;
+      sha256 = "09y4x1yaqm84w7cy1sf57a6g8ah49b0349dsj2a81wy4flw8ih5j";
+    })
+    ];
```
Comment (jtojnar) [inline pkgs/development/tools/misc/elfutils/default.nix]: Please dedent this.

File: pkgs/development/tools/misc/elfutils/default.nix (hunk 2 above)
Comment (jtojnar) [inline pkgs/development/tools/misc/elfutils/default.nix]: Maybe keep the source comment, or mention that it is for GCC 8, or something

**Review by jtojnar — APPROVED** [review]

**Pull request merged**